src/java.desktop/share/classes/java/beans/PropertyChangeSupport.java

```diff
@@ -353,7 +353,7 @@ public void addPropertyChangeListener(
         }
     }
 
-    private static void fire(@Nullable PropertyChangeListener[] listeners, PropertyChangeEvent event) {
+    private static void fire(PropertyChangeListener @Nullable [] listeners, PropertyChangeEvent event) {
```

Comment:
Why is @Nullable after "PropertyChangeListener"? The array of listeners can be null, so I would have through the type name is "@Nullable PropertyChangeListener".

Reply:
Thanks for the followup.

The reason is that a type annotation modifies the (part of the) type that immediately follows it.

`@Nullable PropertyChangeListener[] listeners` means "non-null array of possibly-null PropertyChangeListener".

`PropertyChangeListener @Nullable [] listeners` means "possibly-null array of non-null PropertyChangeListener"

This is also discussed at https://checkerframework.org/manual/#faq-array-syntax-meaning .

Reply:
Got it.